server/database/Models/User.js

```diff
@@ -0,0 +1,90 @@
+
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const mongoose = require('../connection');
+
+const { JWT_SECRET } = process.env;
+
+const userSchema = mongoose.Schema({
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  username: {
+    type: String,
+    required: true,
+    trim: true,
+    unique: true,
+    minlength: 3,
+  },
+  password: {
+    type: String,
+    required: true,
+    minlength: 6,
+  },
+  tokens: [
+    {
+      access: {
+        type: String,
+        required: true,
+      },
+      token: {
+        type: String,
+        required: true,
+      },
+    },
+  ],
+});
+
+/**
+ * Overides the model's default JSON stringify method
+ * @returns Object with user ID and username
+ */
+userSchema.methods.toJSON = function toJSON() {
+  const user = this;
+  /* eslint-disable-next-line no-underscore-dangle */
+  return { _id: user._id, username: user.username };
+};
+
+/**
+ * Generates new JWT and saves in database
+ * @returns Promise with token value
+*/
+userSchema.methods.generateAuthToken = function generateAuthToken() {
+  const user = this;
+  const access = 'auth';
+  const token = jwt.sign({
+    /* eslint-disable-next-line no-underscore-dangle */
+    _id: user._id,
+    username: user.username,
+    access,
+  }, JWT_SECRET);
+
+  user.tokens.push({
+    access,
+    token,
+  });
+  return user.save().then(() => token);
```

Comment:
Why are you saving tokens in database?

Reply:
The user will pass this token when making API calls. If we don't have the tokens on the server, we won't be able to verify them. Do you have a better approach?

Reply:
Yeah, we can use `jwt.verify` to verify it and [jwt-decode](https://github.com/auth0/jwt-decode) to get user from it. 

Or even better, there is an express middleware that does it, take a look at this one.  https://www.npmjs.com/package/express-jwt

Reply:
Saving tokens is a bad idea for the same reason saving plaintext passwords is, if someone gets access to the db, he can login as any user.

Reply:
Okay, got it. Thank you. I will make this change.